components/datetime/src/format/datetime.rs

```diff
@@ -533,28 +533,23 @@ mod tests {
         use icu_provider::prelude::*;
 
         let provider = icu_testdata::get_provider();
-        let locale: Locale = "en-u-ca-gregory".parse().unwrap();
-        let date_data: DataPayload<DateSymbolsV1Marker> = provider
-            .load(&DataRequest {
-                locale: locale.clone().into(),
-                metadata: Default::default(),
-            })
-            .unwrap()
-            .take_payload()
-            .unwrap();
-        let time_data: DataPayload<TimeSymbolsV1Marker> = provider
-            .load(&DataRequest {
-                locale: locale.clone().into(),
-                metadata: Default::default(),
-            })
-            .unwrap()
-            .take_payload()
-            .unwrap();
+        let locale = "en-u-ca-gregory".parse::<Locale>().unwrap();
```

Comment:
`nit`: I think the previous notation is easier to visually parse. Are you opinionated here or is it just an artifact?

Reply:
what notation exactly?

Reply:
```rust
let locale: Locale = "foo".parse().unwrap();
```

Reply:
ah, I much prefer `.parse::<>` because then the expression has a type and you can move it around without changes.